CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.1)
+cmake_minimum_required (VERSION 3.18.6)
```

Comment:
the appveyor CI (for windows) has cmake 3.16.2, so this is why those fail ... and we probably don't want to raise the required cmake version unnecessarily, for example ubuntu 16.04 has cmake 3.5.1

Reply:
`check_linker_flag` was introduced in cmake 3.18 in may 2020.

Well I think that distros should keep cmake updated.

You want to update your CI or want me to provide some primitives as replacement for that cmake module?

Reply:
Well, it's not my project.

But, this seems a good argument to just use the linker script which works with all linkers, so this CMakeLists will remain compatible with linux distros and embedded build toolchains which most people still use.

Reply:
Fixed